src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -824,6 +826,15 @@ private async Task<T> ExecuteWithFailOverPolicyAsync<T>(IEnumerable<Configuratio
 
                 try
                 {
+                    Uri currentEndpoint = _configClientManager.GetEndpointForClient(currentClient);
+
+                    if (failureOccurred && previousEndpoint != currentEndpoint)
```

Comment:
We dont need to maintain `failureOccurred` variable. If `previousEndpoint != currentEndpoint`, we know that failure must've occurred.